### ISSUE TITLE
Osara sometimes crippling UIA with NVDA

### ISSUE THREAD

**Issue by pitermach** [jcsteh/osara#389]
Sorry if this is a duplicate, I spent the last 5 minutes searching for an issue related to this because I know it was mentioned somewhere but I didn't find one. If I overlooked something feel free to close as a duplicate.

This is related to the problem that sometimes Osara refuses to speak with NVDA. With the latest version, osara_2020.1pre-333,29a87ac2, I still experienced this problem after starting Reaper. Something interesting I noticed when this happened though is that before I restarted NVDA and was using the modern Windows 10 Mail app, certain announcements, like the ones indicating the group and list names, stopped working and that object navigation in UWP apps also broke. With NVDA on info logging level I didn't get anything, but when set on debug attempting to use object navigation in the mail app gives this:
DEBUGWARNING - scriptHandler.executeScript (18:13:01.225) - MainThread (33180):
error executing script: <bound method GlobalCommands.script_navigatorObject_previous of <globalCommands.GlobalCommands object at 0x0706FA50>> with gesture 'NVDA+numpad 4'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 208, in executeScript
  File "globalCommands.pyc", line 994, in script_navigatorObject_previous
  File "api.pyc", line 250, in setNavigatorObject
  File "NVDAObjects\__init__.pyc", line 384, in _get_treeInterceptor
  File "treeInterceptorHandler.pyc", line 24, in getTreeInterceptor
  File "UIABrowseMode.pyc", line 478, in __contains__
  File "comtypesMonkeyPatches.pyc", line 26, in __call__
_ctypes.COMError: (-2147220991, 'An event was unable to invoke any of the subscribers', (None, None, None, 0, None))

Finally, this issue doesn't happen just when starting Reaper. I was in a long recording session yesterday, (2+ hours), while using my computer at the time to check emails and look at a few websites. When I started recording, I had speech in Reaper, but at some point during the recording process something broke and Osara had stopped speaking. I experienced such a situation a couple of times, though this was before I updated to the latest Osara build. So this issue may be Osara related or perhaps it could be something else in the system causing UIA to cripple, I don't know if this was at least somewhat helpful or made the situation even more confusing. :)

**Comment by jcsteh** [comment]
I've never been able to reproduce this and those that can reproduce it can't do so reliably, which makes this incredibly difficult to diagnose. My only guess so far is that REAPER's main thread gets blocked sometimes (e.g. if you're performing some complex operation), triggering UIA's connection recovery behaviour (because UIA thinks the app is misbehaving). However, I've tried simulating that by blocking the main thread for 5 seconds after raising a UIA notification event and it doesn't cause UIA to break for me. So, I don't think that's the cause.

A few questions:

1. Do you have "Report changes made via control surfaces" enabled in OSARA Configuration? Does this happen with it disabled?
2. Do you have any NVDA add-ons installed? Does this still happen if you disable them?

**Comment by ScottChesworth** [comment]
We recently had a chap on RWP who traced a similar sounding situation to be caused by the presence of Win10 App Essentials NVDA addon on his machine. Do you have that installed, per chance @pitermach?

**Comment by pitermach** [comment]
In answer to both of your questions, yes I have the report control surface changes option checked. For ad-ons, yes I have way too many. The ones that would be the most likely to have an effect because they're global plugins I have BrailleExtender, Tony's enhancements and Windows 10 app essentials. That last one I thought I had enabled, but as it turns out it wasn't. I just enabled it and had a long session of using Reaper and everything worked great for hours. I also just disabled Tony's enhancements since I'm not even sure what I was using it for. I know none of this is probably too helpful, I'll leave this open for now and will let you know if the issue comes up again or someone else sees something we're not.
